### PR TITLE
[REF] mrp_bom_hierarchy with OCA guidelines

### PR DESCRIPTION
**Pull request by quentinDupont**
- one model.py per model
- rename view file
- rename compute methods as `_compute_field_name`

Thanks for this module !

### CODE REVIEW

**Comment by quentinDupont** [comment]
waiting for this PR https://github.com/OCA/manufacture/pull/1205/files before rebase this one

**Review by legalsylvain — APPROVED** [review]
Thanks !

**Comment by quentinDupont** [comment]
ping @JordiBForgeFlow @MiquelRForgeFlow :)

File: mrp_bom_hierarchy/README.rst [hunk 1]
```diff
@@ -62,6 +62,7 @@ Authors
 ~~~~~~~
 
 * ForgeFlow
+* Grap
 
 Contributors
```
Comment (MiquelRForgeFlow) [inline mrp_bom_hierarchy/README.rst]: I think there is a difference between author and contributor, and I think what you did is a contribution. What do you think?

**Review by MiquelRForgeFlow — COMMENTED** [review]
Nice contribution! :)

File: mrp_bom_hierarchy/README.rst (hunk 1 above)
Comment (quentinDupont) [reply mrp_bom_hierarchy/README.rst]: yes totally agree with you ! i change it  !

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by legalsylvain** [comment]
@MiquelRForgeFlow : could you merge this approved one ? and also this trivial one here : https://github.com/OCA/manufacture/pull/1205#pullrequestreview-2121594507

thanks !

**Comment by rvalyi** [comment]
cc @renatonlima

File: mrp_bom_hierarchy/__manifest__.py [hunk 2]
```diff
@@ -6,12 +6,13 @@
     "summary": "Make it easy to navigate through BoM hierarchy.",
     "version": "16.0.1.0.1",
     "author": "ForgeFlow, Odoo Community Association (OCA)",
+    "contributors": "Grap",
```
Comment (LoisRForgeFlow) [inline mrp_bom_hierarchy/__manifest__.py:9]: does this key even exist?

Comment (MiquelRForgeFlow) [reply mrp_bom_hierarchy/__manifest__.py:9]: Yes, here https://github.com/odoo/odoo/blob/16.0/odoo/addons/base/models/ir_module.py#L729

Comment (LoisRForgeFlow) [reply mrp_bom_hierarchy/__manifest__.py:9]: interesting, TIL, thanks!

File: mrp_bom_hierarchy/model/mrp_bom.py [hunk 3]
```diff
@@ -164,105 +164,3 @@ def action_open_product_other_bom_tree_view(self):
...
-    def _compute_child_bom_id(self):
-        res = super()._compute_child_bom_id()
-        for line in self:
-            line.has_bom = bool(line.child_bom_id)
-        return res
-
-    def action_open_product_bom_tree_view(self):
-        self.ensure_one()
-        res = self.env["ir.actions.actions"]._for_xml_id("mrp.mrp_bom_form_action")
-        if len(self.child_bom_id) == 1:
```
Comment (hparfr) [inline mrp_bom_hierarchy/model/mrp_bom.py:187]: @quentinDupont 

Was it on purpose to remove this check ?

Comment (quentinDupont) [reply mrp_bom_hierarchy/model/mrp_bom.py:187]: damn i don't remember sorry

Comment (hparfr) [reply mrp_bom_hierarchy/model/mrp_bom.py:187]: I have restored it in v18: https://github.com/OCA/manufacture/pull/1551

**Pull request merged**